lib/mongodb.js

```diff
@@ -640,10 +640,39 @@ MongoDB.prototype.update =
       debug('updateAll', model, where, data);
     }
     var idName = this.idName(model);
-    delete data[idName];
 
     where = self.buildWhere(model, where);
-    this.collection(model).update(where, {$set: data}, {multi: true, upsert: false},
+    var parsedData = {};
+    delete data[idName];
+
+    if (self.settings.allowExtendedOperators === true) {
+      var acceptedOperators = [
+        // Field operators
+        '$currentDate', '$inc', '$max', '$min', '$mul', '$rename', '$setOnInsert', '$set', '$unset',
+        // Array operators
+        '$addToSet', '$pop', '$pullAll', '$pull', '$pushAll', '$push',
+        // Bitwise operator
+        '$bit'
+      ];
```

Comment:
Maybe even better for `allowExtendedOperators` to be `true` or an array of operator names.